accepted/2024/release-staging-and-deprecation.md

```diff
@@ -0,0 +1,202 @@
+
+## Prior Art
+
+<!-- What prior art, both good and bad are related to this proposal? -->
+<!-- Do other features exist in other ecosystems and what experience have their community had? -->
+<!-- What lessons from other communities can we learn from? -->
+<!-- Are there any resources that are relevant to this proposal? -->
+
+Below are two problems from Microsoft, where due to lack of this feature teams are struggling and have to come up with potential alternative solutions.
+
+### .NET Release publishing
+
+The .NET team has a monthly release cadance for their products. As part of each release we publish hundreds of packages to NuGet.
+This is generally a very time-sensitive effort, especially when it comes to security releases. Many teams within Microsoft
+ wait for our releases before they can move forward with publishing their patches immediately after, because with security releases
+ we will also document the security fixes the release contains. Unfortunately, what happens sometimes, is that
+ during publishing NuGet sometimes blocks certain packages due to some new validation rules or some other reason, which
+ wasn't something the team could have foreseen earlier. As a result, the the releases get delayed and partner teams delay
+ their releases awaiting a green light from the .NET team, which results in a tense / stressfull effort of patching the impacting
+ packages on the last minute to address the faced issues.
+Besides the security releases, there are also GA releases, which have a similar risk characteristics.
+Many times there are conferences scheduled the same day when a release is planned. And during such a conference the presenter
+ expectes the release to be published (a few hours earlier) so that they can demo some new feature(s).
+ And it happened in the past that the releases didn't succeed in time and impacted the presentations in a negative way.
+ expectes the release to be published (a few hours earlier) so that they can demo some new feature(s).
+ And it happened in the past that the releases didn't succeed in time and impacted the presentations in a negative way.
+
+Having a mechanism to stage / pre-publish packages ahead of time and then only to `notify` NuGet to make a release available
+ would reduce such risks dramatically, as issues would have been caught earlier during package staging,
+ which can happen days before the actual scheduled release.
+
+### .NET release deprecation
+As part of the .NET product support policy, releases go out of support after a certain period of time. When that happens, the team has some automation to "find out" what were all the packages that were released as part of the release and then utilizes a custom built solution / tool to deprecate all these packages by calling multiple NuGet APIs per each package to do so. This is also a stressful process, such, that we had to introduce some artificial delays in the code to avoid stressing the servers.
```

Comment:
Does nuget.org have a private API to deprecate packages? FYI, the Azure SDK team had requested a similar API to manage their packages: https://github.com/NuGet/NuGetGallery/issues/8873